File: csharp_body_deref/Gisketch.Aottg2UI.Runtime/Gisketch.Aottg2UI.Actions/GisketchActionContext.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Gisketch.Aottg2UI.Actions.GisketchActionContext
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Gisketch/Aottg2UI/Actions/GisketchActionContext.c
// Prior real C# source: none (type is newer than the last public source snapshot)
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Gisketch.Aottg2UI.Data;
using Il2CppDummyDll;
using UnityEngine;

namespace Gisketch.Aottg2UI.Actions;

[Token(Token = "0x20000D6")]
public readonly struct GisketchActionContext
{
	[Token(Token = "0x400042A")]
	[FieldOffset(Offset = "0x0")]
	public readonly string ElementId;

	[Token(Token = "0x400042B")]
	[FieldOffset(Offset = "0x8")]
	public readonly GameObject GameObject;

	[Token(Token = "0x400042C")]
	[FieldOffset(Offset = "0x10")]
	public readonly GisketchNodeDefinition Node;

	[Token(Token = "0x400042D")]
	[FieldOffset(Offset = "0x18")]
	public readonly GisketchActionInputSource InputSource;

	[Token(Token = "0x60005F8")]
	[Address(RVA = "0x3A9D200", Offset = "0x3A9D200", VA = "0x3A9D200")]
	public GisketchActionContext(string elementId, GameObject gameObject, GisketchNodeDefinition node)
	{
	}

	[Token(Token = "0x60005F9")]
	[Address(RVA = "0x3A9D250", Offset = "0x3A9D250", VA = "0x3A9D250")]
	public GisketchActionContext(string elementId, GameObject gameObject, GisketchNodeDefinition node, GisketchActionInputSource inputSource)
	{
	}
}

File: cpp/Gisketch/Aottg2UI/Actions/GisketchActionContext.c
// Type: Gisketch.Aottg2UI.Actions.GisketchActionContext
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Gisketch.Aottg2UI.Runtime/Gisketch.Aottg2UI.Actions/GisketchActionContext.cs
// Prior real C# source: none
// --------------------------------

// Gisketch.Aottg2UI.Actions.GisketchActionContext$$.ctor
// il2cpp: void Gisketch_Aottg2UI_Actions_GisketchActionContext___ctor (Gisketch_Aottg2UI_Actions_GisketchActionContext_o __this, System_String_o* elementId, UnityEngine_GameObject_o* gameObject, Gisketch_Aottg2UI_Data_GisketchNodeDefinition_o* node, const MethodInfo* method);
// 0x3b9d200

void Gisketch_Aottg2UI_Actions_GisketchActionContext___ctor
               (Gisketch_Aottg2UI_Actions_GisketchActionContext_o __this,System_String_o *elementId,
               UnityEngine_GameObject_o *gameObject,Gisketch_Aottg2UI_Data_GisketchNodeDefinition_o *node,
               MethodInfo *method)

{
  elementId->klass = (System_String_c *)gameObject;
  il2cpp_runtime_helper_022b4080();
  elementId->monitor = node;
  il2cpp_runtime_helper_022b4080(&elementId->monitor,node);
  elementId->fields = (System_String_Fields)method;
  il2cpp_runtime_helper_022b4080(&elementId->fields,method);
  *(undefined4 *)&elementId[1].klass = 0;
  return;
}


// Gisketch.Aottg2UI.Actions.GisketchActionContext$$.ctor
// il2cpp: void Gisketch_Aottg2UI_Actions_GisketchActionContext___ctor (Gisketch_Aottg2UI_Actions_GisketchActionContext_o __this, System_String_o* elementId, UnityEngine_GameObject_o* gameObject, Gisketch_Aottg2UI_Data_GisketchNodeDefinition_o* node, int32_t inputSource, const MethodInfo* method);
// 0x3b9d250

void Gisketch_Aottg2UI_Actions_GisketchActionContext___ctor_3a9d250
               (Gisketch_Aottg2UI_Actions_GisketchActionContext_o __this,System_String_o *elementId,
               UnityEngine_GameObject_o *gameObject,Gisketch_Aottg2UI_Data_GisketchNodeDefinition_o *node,
               int32_t inputSource,MethodInfo *method)

{
  System_String_Fields SVar1;
  undefined4 in_register_0000000c;
  
  SVar1._4_4_ = in_register_0000000c;
  SVar1._stringLength = inputSource;
  elementId->klass = (System_String_c *)gameObject;
  il2cpp_runtime_helper_022b4080();
  elementId->monitor = node;
  il2cpp_runtime_helper_022b4080(&elementId->monitor,node);
  elementId->fields = SVar1;
  il2cpp_runtime_helper_022b4080(&elementId->fields,SVar1);
  *(int *)&elementId[1].klass = (int)method;
  return;
}


